inc/cppxaml/XamlWindow.h

```diff
@@ -0,0 +1,256 @@
+                ShowWindow(hWnd, nCmdShow);
+                UpdateWindow(hWnd);
+            }
+            return hWnd;
+        }
+
+        void SetAcceleratorTable(HACCEL acc) {
+            hAccelTable = acc;
+        }
+
+        AppController* Controller() const {
+            return controller;
+        }
+
+        int RunLoop() {
+            MSG msg;
+
+            // Main message loop:
+            while (GetMessage(&msg, nullptr, 0, 0))
+            {
+                if (auto xamlSourceNative2 = desktopXamlSource.as<IDesktopWindowXamlSourceNative2>()) {
+                    BOOL xamlSourceProcessedMessage = FALSE;
+                    winrt::check_hresult(xamlSourceNative2->PreTranslateMessage(&msg, &xamlSourceProcessedMessage));
+                    if (xamlSourceProcessedMessage) {
+                        continue;
+                    }
+                }
+
+                if (!TranslateAccelerator(msg.hwnd, hAccelTable, &msg))
+                {
+                    TranslateMessage(&msg);
+                    DispatchMessage(&msg);
+                }
+            }
+            return (int)msg.wParam;
+        }
+
+        HWND GetBridgeWindow() const {
+            auto interop = desktopXamlSource.as<IDesktopWindowXamlSourceNative>();
+            HWND hWndXamlIsland = nullptr;
+            winrt::check_hresult(interop->get_WindowHandle(&hWndXamlIsland));
+            return hWndXamlIsland;
+        }
+
+        LRESULT WndProc(HWND hwnd, UINT message, WPARAM wParam, LPARAM lParam) {
+            auto interop = desktopXamlSource.as<IDesktopWindowXamlSourceNative>();
+
+            switch (message)
+            {
+            case WM_CREATE: {
+                this->hWnd = hwnd;
+                auto createStruct = reinterpret_cast<LPCREATESTRUCT>(lParam);
+
+                // Parent the DesktopWindowXamlSource object to the current window.
+                winrt::check_hresult(interop->AttachToWindow(hWnd));  // This fails due to access violation!
+
+                auto hWndXamlIsland = GetBridgeWindow();
+                SetWindowPos(hWndXamlIsland, nullptr, 0, 0, createStruct->cx, createStruct->cy,
+                    SWP_SHOWWINDOW | SWP_NOACTIVATE);
+
+                auto mainPage = this->getUI(*this);
+                if (controller && controller->OnUICreated) {
+                    controller->OnUICreated(mainPage, this);
+                }
+                desktopXamlSource.Content(mainPage);
+                break;
+            }
+            case WM_SIZE:
+            {
+                HWND hWndXamlIsland = nullptr;
+                winrt::check_hresult(interop->get_WindowHandle(&hWndXamlIsland));
+
+                SetWindowPos(hWndXamlIsland, nullptr, 0, 0, LOWORD(lParam), HIWORD(lParam), SWP_SHOWWINDOW);
+
+                break;
+            }
+            }
+            if (controller && controller->WndProc) {
+                return controller->WndProc(hWnd, message, wParam, lParam, this);

```

Comment:
seems you need to enable per window storage instead of using controller as a global. this is to enable multi-window app.